media-api/app/lib/UsageStore.scala

```diff
@@ -59,29 +59,67 @@ object StoreAccess {
   implicit val writes: Writes[StoreAccess] = Json.writes[StoreAccess]
 }
 
+object UsageStore {
+  def extractEmail(stream: InputStream): List[String] = {
+    val s = Session.getDefaultInstance(new Properties())
+    val message = MimeMessageUtils.createMimeMessage(s, stream)
+
+    message.getContent match {
+      case content: MimeMultipart =>
+        val count = content.getCount
+
+        var list = List.empty[String]
+
+        for(n <- 0 until count) {
+          content.getBodyPart(n) match {
+            case part: MimeBodyPart if part.getEncoding == "base64" =>
+              part.getContent match {
+                case c: InputStream => list = scala.io.Source.fromInputStream(c).getLines().toList
+              }
+            case _ =>
+          }
+        }
+        list
+    }
+  }
+
+  def csvParser(list: List[String]): List[SupplierUsageSummary] = {
+    def stripQuotes(s: String): String = s.stripSuffix("\"").stripPrefix("\"")
+
+    list.headOption match {
+      case Some(head) if head.split(',').toList.length == 2 =>
+        list.tail.map { line =>
+          val lineList = line.split(',')
+          val (supplier, count) = (
+            stripQuotes(lineList.head),
+            stripQuotes(lineList.tail.head)
```

Comment:
we could also check the values of the header row, to further ensure its the right csv we're looking for